src/stream/stream.client.ts

```diff
@@ -199,8 +200,15 @@ export class StreamClient extends EventEmitter<StreamClientEvents> {
    *
    * @param {WebSocket.Data} data
    */
-  private onMessage(data: Data): void {
-    if (!(data instanceof Buffer)) {
+  private onMessage(event: WebSocket.MessageEvent): void {
+    const { data } = event;
+
+    let parse: (data: any) => CensusMessage;
+    if (typeof data === 'string') {
+      parse = (data: string) => JSON.parse(data);
+    } else if (data instanceof Buffer) {
+      parse = (data: Buffer) => JSON.parse(data.toString());
+    } else {
```

Comment:
Maybe just extend the guard as you lose some type safety now. Using `toString` on a string gives the same result anyway.
```suggestion
    if (typeof data !== 'string' && !(data instanceof Buffer)) {
```

Reply:
`Buffer` isn't defined in browser, so will have to handle that a bit more carefully to avoid an early error-out.